x/blockdb/database.go

```diff
@@ -1134,14 +1149,36 @@ func (s *Database) writeBlockAt(offset uint64, bh blockEntryHeader, block BlockD
 	combinedBuf := make([]byte, combinedBufSize)
 	copy(combinedBuf, headerBytes)
 	copy(combinedBuf[sizeOfBlockEntryHeader:], block)
-	if _, err := dataFile.WriteAt(combinedBuf, int64(localOffset)); err != nil {
-		return fmt.Errorf("failed to write block to data file at offset %d: %w", offset, err)
-	}
+	dataWritten := false
 
-	if s.config.SyncToDisk {
-		if err := dataFile.Sync(); err != nil {
-			return fmt.Errorf("failed to sync data file after writing block %d: %w", bh.Height, err)
+	for {
+		dataFile, localOffset, fileIndex, err := s.getDataFileAndOffset(offset)
+		if err != nil {
+			return fmt.Errorf("failed to get data file for writing block %d: %w", bh.Height, err)
+		}
+
+		if !dataWritten {
```

Comment:
why do we skip overwriting? 

If we have pending bytes to be written to the file but we closed the file, are we guaranteed that if we re-open the file and call sync, the changes from before will be written? 

Isn't that dependent on the implementation of the file handle? The file handle in the next iteration isn't the same as the one in the previous one.

Can't we just overwrite the changes again? we're not appending to the file but writing to a specific offset, right?  Then the code will also be simpler.